### PR TITLE
Feature engineering and other stuff

### PR DESCRIPTION
**Pull request by PrimeF**
- Added gitignore entries for model weights, experiment runs and tensorboard events (in the future we should create a **_production_** directory which we actually sync with git and where we keep our best model)
- Created a first version of feature engineering. Still WIP!
- Fix model fit function
- Introduce experiments tracking functionality
- Changed settings.py for task_1. Change back when submitting!

### CODE REVIEW

File: agent_code/auto_bomber/train.py [hunk 1]
```diff
@@ -79,13 +79,25 @@ def reward_from_events(self, events: List[str]) -> int:
...
+        e.COIN_COLLECTED: 100,
+        e.KILLED_OPPONENT: 50,
+        e.INVALID_ACTION: -100,
+        e.KILLED_SELF: -300,
+        e.GOT_KILLED: -50,
+        e.WAITED: -10,
+        e.SURVIVED_ROUND: 5
     }
+
+    # game_rewards = {
```
Comment (tkrieger) [inline agent_code/auto_bomber/train.py]: could be removed completely or?

File: .gitignore [hunk 2]
```diff
@@ -114,3 +114,4 @@ replays/
 agent_code/auto_bomber/models
 agent_code/auto_bomber/*.pt
 agent_code/auto_bomber/runs
+.DS_Store
```
Comment (tkrieger) [inline .gitignore]: empty line ;)

**Review by tkrieger — COMMENTED** [review]
Only two minor things, no re-review required from my side after fixing.

I did not add comments for feature_engineering.py as WIP. Nevertheless I had a close look and think the idea is great and should work out quite will.
As you not adapted the tests, and they are anyway not working any longer, you can even delete them ;)

File: agent_code/auto_bomber/model.py [hunk 3]
```diff
@@ -9,25 +10,42 @@
...
-        path = Path(config.MODEL_PATH)
-        if path.is_file():
-            with path.open(mode="rb") as file:
+        if config.MODEL_DIR and Path(config.MODEL_DIR).is_dir():
+            self.model_dir = Path(config.MODEL_DIR)
+        elif config.MODEL_DIR and not Path(config.MODEL_DIR).is_dir():
+            raise FileNotFoundError("The specified model directory does not exist!\nIf you wish to train a NEW model"
+                                    "set parameter to None, otherwise specify a valid model directory.")
+        else:
+            model_index = sorted([int(x.stem) for x in Path(config.MODELS_ROOT).iterdir() if x.is_dir()])[-1]
```
Comment (tkrieger) [inline agent_code/auto_bomber/model.py:26]: this will fail on a empty directory

File: agent_code/auto_bomber/train.py (hunk 1 above)
Comment (PrimeF) [reply agent_code/auto_bomber/train.py]: Sure!

File: .gitignore (hunk 2 above)
Comment (PrimeF) [reply .gitignore]: Thanks

**Comment by PrimeF** [comment]
Cleaned up, changed feature engineering and some folder handling.
@tkrieger @GucciGui: A review is appreciated.

File: agent_code/auto_bomber/model.py (hunk 3 above)
Comment (tkrieger) [reply agent_code/auto_bomber/model.py:26]: will still fail due to index out of bound - I fixed it in my version? Shall I simply put my changes in here?

**Review by GucciGui — APPROVED** [review]

**Comment by PrimeF** [comment]
**_Please do not delete branch!_**

**Pull request merged**